draft-ietf-rats-eat.md

```diff
@@ -1136,12 +1136,17 @@ The following sub-sections define the three types for representing submodules:
 
 When decoding a submodule in a CBOR-encoded EAT, the Claims-Set type will be encoded as a map, the Detached-Submodule-Digest type as an array, and the Nested-Token type as a CBOR-tagged object.
 
-When decoding a JSON-encoded EAT, a little more work is required because both the Nested-Token and Detached-Submodule-Digest types are arrays.
-To distinguish the nested token from the detached digest, the first element in the array is examined.
-If it is "JWT" or "BUNDLE", then the submodule is a Nested-Token.
-Otherwise it will contain an algorithm identifier and is a Detached-Submodule-Digest.
-
-Where Nested-Token is used, nested CBOR EATs MUST be tagged, i.e., a CBOR tag will be used to distinguish between CWT or BUNDLE. Nested JSON EATs will be encoded as an JSON-Nested-Token, with the type indicated in the type field, i.e., first element in the array. The string identifying the JSON-encoded token MUST be one of the following:
+When decoding a JSON-encoded EAT submodule, the Claims-Set type is encoded in a map and all the other types an array.
+The first member of the array gives further info.
+If it is an algorithm identifier it is a detached digest.
+If it is "JWT", "CBOR" or "BUNDLE", it is a Nested-Token in the indicated form.
+
+When decoding a Nested-Token, if a byte string is encountered, the nested token is a CBOR-encoded token.
+The byte string wrapping is removed and the type of the token is determined by the CBOR tag.
```

Comment:
There is no byte string wrapping for CBOR-encoded tokens in a JSON surrounding token.

Think its gonna be hard to have common text for Nested-Token for both CBOR and JSON because they are very different. It is one of the two places where the CDDL can't be the same for JSON and CBOR.

I do like the general structure where some prescription for decoding is given.

Reply:
I moved the sentence addressing CBOR to be the first sentence of this section (so these words are now CBOR-specific).